### PR TITLE
[SDP-852] Receiver registration: phone number input with countries dropdown

### PR DESCRIPTION
**Pull request by quietbits**
### What

Updated phone number input to show countries dropdown and validate the number for better UX. Jira ticket: https://stellarorg.atlassian.net/browse/SDP-852

image

image

### Why

It was not a good user experience to ask users to input the phone number in a specific format without any spaces, dashes, etc.

### Known limitations

N/A

#### PR Structure

* [x] This PR has reasonably narrow scope (if not, break it down into smaller PRs).
* [x] This PR does not mix refactoring changes with feature changes (split into two PRs otherwise).
* [x] This PR's title starts with the name of the package, area, or subject affected by the change.

#### Thoroughness

* [x] This PR adds tests for the new functionality or fixes.
* [x] This PR contains the link to the Jira ticket it addresses.

#### Configs and Secrets

* [x] No new CONFIG variables are required -OR- the new required ones were added to the helmchart's [`values.yaml`] file.
* [x] No new CONFIG variables are required -OR- the new required ones were added to the deployments ([`pr-preview`], [`dev`], [`demo`], `prd`).
* [x] No new SECRETS variables are required -OR- the new required ones were mentioned in the helmchart's [`values.yaml`] file.
* [x] No new SECRETS variables are required -OR- the new required ones were added to the deployments ([`pr-preview secrets`], [`dev secrets`], [`demo secrets`], `prd secrets`).

#### Release

* [x] This is not a breaking change.
* [x] **This is ready for production.**. If your PR is not ready for production, please consider opening additional complementary PRs using this one as the base. Only merge this into `develop` or `main` after it's ready for production!

#### Deployment

* [ ] Does the deployment work after merging?

[`values.yaml`]: ../helmchart/sdp/values.yaml
[`pr-preview`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/common-previews/stellar-disbursement-platform/backend-helm-values
[`dev`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/stellar-disbursement-platform/backend-helm-values
[`demo`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/stellar-disbursement-platform/demo/demo-backend-helm-values
[`pr-preview secrets`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/common-previews/externalsecrets-common-previews.yaml#L241-L346
[`dev secrets`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/stellar-disbursement-platform/stellar-disbursement-platform-externalsecrets.yaml
[`demo secrets`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/stellar-disbursement-platform/demo/demo-sdp-externalsecrets.yaml

### CODE REVIEW

**Review by marcelosalloum — APPROVED** [review]
LGTM!

**Comment by marcelosalloum** [comment]
I've tested it and it worked great. I haven't reviewed the code thoroughly though. Maybe you'd want a pass of @ceciliaromao before merging it.

**Comment by marcelosalloum** [comment]
@quietbits: one small suggestion could be to set the starting country based on some IP information from the user, but that might be too much work, so I'm not sure it's worth it right now.

**Comment by quietbits** [comment]
> @quietbits: one small suggestion could be to set the starting country based on some IP information from the user, but that might be too much work, so I'm not sure it's worth it right now.

@marcelosalloum This library might actually support that. I'll try that.

File: internal/htmltemplate/tmpl/receiver_register.tmpl
Comment (piyalbasu) [inline internal/htmltemplate/tmpl/receiver_register.tmpl:20]: I think it might be better to link to a cdn hosted version of the CSS and JS. You're probably trying to avoid additional roundtrips? It makes me a little nervous to manually add these lib files into the repo 😅 

Not sure how much of a lift it would be to use npm for this (I'm assuming significant), but I think if you're going to use an external lib, prob best to use either a cdn or a package manager

Comment (marcelosalloum) [reply internal/htmltemplate/tmpl/receiver_register.tmpl:20]: The security implication here is that we use Content-Security-Policy to allowlist specific domains where we will fetch resources. So if we want to use CDN, we just need to whitelist the address.

Is the CDN approach a recommendation for improved performance? Or is there another reason? If performance is the only factor, we could merge this PR and open a new one where we adopt the CDN, right?

Comment (quietbits) [reply internal/htmltemplate/tmpl/receiver_register.tmpl:20]: I don't think we'd gain anything in performance by using CDN. This will be the only page where these files are used, so caching locally vs from CDN wouldn't make much difference. It would be more convenient to upgrade to a newer version, though. By hosting the files ourselves, we're guaranteed that it will work if the page works. CDN might have a blackout, but this is probably unlikely.

I'd say let's merge this in and see how it goes. If we need to move to CDN, we can make the switch later. How do you feel about that, @piyalbasu @marcelosalloum ?

Comment (piyalbasu) [reply internal/htmltemplate/tmpl/receiver_register.tmpl:20]: I'm okay with merging this as is. My real recommendation would be to use npm or a package manager to handle external libs, but a CDN is probably less of a lift here. 

Locally hosting a versioned library just tends to be untenable in the long run and it is a bit of an antipattern. It adds friction when upgrading and adds unnecessary bloat to a codebase. Also, in my own experience with hosting code locally like this, someone always ends up directly editing the library files which can really cause headaches 😄 .

But none of these are dealbreakers to me. If you all are okay with taking on that cost, that's totally your call!

**Review by piyalbasu — APPROVED** [review]

**Review by ceciliaromao — COMMENTED** [review]
LGTM! 🚀

**Review by ceciliaromao — APPROVED** [review]

File: internal/htmltemplate/tmpl/receiver_register.tmpl
Comment (marcelosalloum) [reply internal/htmltemplate/tmpl/receiver_register.tmpl:20]: I agree this is not ideal for application development, but considering this was meant to be a lightweight application that is used to get PII of recipients in a secure manner, we are weighing towards this kind of solution. That perception can change if we decide to increase the complexity of these pages, and if that's the case, we should definitely flag it.

> someone always ends up directly editing the library files

I agree this would be bad.

---
> a CDN is probably less of a lift in this situation, which is why I mentioned it.

Using CDN/cache would be a deployment-specific approach and every entity hosting the SDP would need to do that independently. We can recommend that in the README.

> My real recommendation would be to use npm or a package manager to handle external libs

Package management is probably a long-term solution and it's something that'd make sense if we decide to increase the complexity of this interface. I believe it could eventually become necessary, but it's not in our foreseeable future timeline yet. If @quietbits or @piyalbasu feel strongly about migrating to a react-app or another approach that uses npm package management, we can bring that up for discussion.

Comment (quietbits) [reply internal/htmltemplate/tmpl/receiver_register.tmpl:20]: We should make it a proper web app if it gets more complex. :100: I don't know what else we're planning to add, but this is probably the max we should do with the current setup.

**Pull request merged**
